### ISSUE TITLE
Postgres docker container not listening

### ISSUE THREAD

**Issue by NachoPicchu** [LemmyNet/lemmy-ansible#190]
Using the 1.2.0 version of the ansible script on a fresh debian 11 installation, the created postgres docker container does not appear to be listening for connections. The playbook runs through without any errors and results in a hosted lemmy page, but that page simply displays an error like here: https://github.com/LemmyNet/lemmy-ansible/issues/141#issuecomment-1663281825

The Docker **Lemmy logs** say:
```
thread 'main' panicked at 'Error connecting to postgres://lemmy:xxxxxxx@postgres:5432/lemmy: could not connect to server: Connection refused
Is the server running on host "postgres" (172.18.0.5) and accepting
TCP/IP connections on port 5432?
', crates/db_schema/src/utils.rs:217:56
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The Docker **Postgres logs** say:
```
PostgreSQL Database directory appears to contain a database; Skipping initialization
2023-10-15 10:34:22.941 GMT [1] LOG:  starting PostgreSQL 15.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit
2023-10-15 10:34:22.942 GMT [1] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2023-10-15 10:34:22.942 GMT [1] LOG:  could not bind IPv6 address "::1": Address not available
2023-10-15 10:34:22.945 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-10-15 10:34:22.953 GMT [24] LOG:  database system was shut down at 2023-10-15 10:34:22 GMT
2023-10-15 10:34:22.964 GMT [1] LOG:  database system is ready to accept connections
2023-10-15 10:39:23.007 GMT [22] LOG:  checkpoint starting: time
2023-10-15 10:39:23.046 GMT [22] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.032 s, sync=0.002 s, total=0.039 s; sync files=2, longest=0.001 s, average=0.001 s; distance=0 kB, estimate=0 kB
```
**docker ps -a** looks like this:
```
CONTAINER ID   IMAGE                        COMMAND                  CREATED        STATUS                            PORTS                                               NAMES
519fed806c20   nginx                        "/docker-entrypoint.…"   12 hours ago   Up 12 hours                       80/tcp, 0.0.0.0:2045->8536/tcp, :::2045->8536/tcp   domainname_proxy_1
8df9a59e604b   asonix/pictrs:0.4.3          "/sbin/tini -- /usr/…"   12 hours ago   Up 12 hours                       6669/tcp, 8080/tcp                                  domainname_pictrs_1
28d86c12f2ef   postgres:15-alpine           "docker-entrypoint.s…"   12 hours ago   Up 15 minutes                     5432/tcp                                            domainname_postgres_1
5fcf8ea7981d   dessalines/lemmy:0.18.5      "/app/lemmy"             13 hours ago   Restarting (101) 16 seconds ago                                                       domainname_lemmy_1
33a140f107e2   dessalines/lemmy-ui:0.18.5   "docker-entrypoint.s…"   13 hours ago   Up 12 hours                       1234/tcp                                            domainname_lemmy-ui_1
b82a8fceaee6   mwader/postfix-relay         "/root/run"              13 hours ago   Up 12 hours                       25/tcp                                              domainname_postfix_1
```

Running **netstat -tulnp**:
Active Internet connections (only servers)
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      4536/nginx: master
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      455/sshd: /usr/sbin
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      4536/nginx: master
tcp        0      0 0.0.0.0:2045            0.0.0.0:*               LISTEN      4329/docker-proxy
tcp6       0      0 :::80                   :::*                    LISTEN      4536/nginx: master
tcp6       0      0 :::22                   :::*                    LISTEN      455/sshd: /usr/sbin
tcp6       0      0 :::443                  :::*                    LISTEN      4536/nginx: master
tcp6       0      0 :::2045                 :::*                    LISTEN      4337/docker-proxy
```
Finally, for good measure **root@machine_xyz:~# docker network inspect domainname_default**:
```
[
{
"Name": "domainname_default",
"Id": "2207f27c9070eeb474d3b82dd0a68da28c38528018cacb9d9692732e4de93e4f",
"Created": "2023-10-14T17:11:22.661151681-04:00",
"Scope": "local",
"Driver": "bridge",
"EnableIPv6": false,
"IPAM": {
"Driver": "default",
"Options": null,
"Config": [
{
"Subnet": "172.18.0.0/16",
"Gateway": "172.18.0.1"
}
]
},
"Internal": false,
"Attachable": true,
"Ingress": false,
"ConfigFrom": {
"Network": ""
},
"ConfigOnly": false,
"Containers": {
"28d86c12f2ef7dffbd82ee32b508a996128d012792c738a8ae241fb25f694b39": {
"Name": "domainname_postgres_1",
"EndpointID": "d9e5359c4c67ff55ca148e028426569573b25aa30b0b7511df83a2ca3616797a",
"MacAddress": "02:42:ac:12:00:05",
"IPv4Address": "172.18.0.5/16",
"IPv6Address": ""
},
"33a140f107e224b2d96119b72de3c6bfbeea1bfca4565b5fc3bdf559d87b62c1": {
"Name": "domainname_lemmy-ui_1",
"EndpointID": "38d71a70d4ee3ccc6eec6f2406809d76741b12f7d8dcf2afb2efa2c23682f344",
"MacAddress": "02:42:ac:12:00:07",
"IPv4Address": "172.18.0.7/16",
"IPv6Address": ""
},
"519fed806c2024f94a2f9dc49f41819141c3a736670be5593eb339d491f148d2": {
"Name": "domainname_proxy_1",
"EndpointID": "9f6a6dfe82d2e4b6b2b0b9588c170992bcd1940a51d758c13152502179426207",
"MacAddress": "02:42:ac:12:00:04",
"IPv4Address": "172.18.0.4/16",
"IPv6Address": ""
},
"8df9a59e604b0701e84a17d4ac238bc925a4e46a6a447490bcb9851fb08c87b4": {
"Name": "domainname_pictrs_1",
"EndpointID": "31a3c59309e29b677599ee38198eade65d101d307ae8594f3036098c987965ff",
"MacAddress": "02:42:ac:12:00:02",
"IPv4Address": "172.18.0.2/16",
"IPv6Address": ""
},
"b82a8fceaee6099893db10d6402c573a3843182c7e4c46c1b313668a89f70d26": {
"Name": "domainname_postfix_1",
"EndpointID": "29a9c1e397d7351a320b4dee8fda54c0724a7f16dbc9a39c2c9a9970ec5b9b4b",
"MacAddress": "02:42:ac:12:00:03",
"IPv4Address": "172.18.0.3/16",
"IPv6Address": ""
}
},
"Options": {},
"Labels": {
"com.docker.compose.network": "default",
"com.docker.compose.project": "domainname",
"com.docker.compose.version": "1.25.0"
}
}
]
```

**Comment by NachoPicchu** [comment]
The reason for the postgres container not listening was that my customPostgresql.conf did not contain the line
`listen_addresses = '*'`
However, **lemmy logs** still show errors related to the DB:
```
thread 'main' panicked at 'Error connecting to postgres://lemmy:xxxxxxxxxx@postgres:5432/lemmy: FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
', crates/db_schema/src/utils.rs:217:56
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Which is mirrored by the **postgres log**:
```
PostgreSQL Database directory appears to contain a database; Skipping initialization

2023-10-15 11:20:57.046 GMT [1] LOG:  starting PostgreSQL 15.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit
2023-10-15 11:20:57.047 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-10-15 11:20:57.047 GMT [1] LOG:  listening on IPv6 address "::", port 5432
2023-10-15 11:20:57.050 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-10-15 11:20:57.057 GMT [24] LOG:  database system was shut down at 2023-10-15 11:20:56 GMT
2023-10-15 11:20:57.066 GMT [1] LOG:  database system is ready to accept connections
2023-10-15 11:21:01.331 GMT [28] FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
2023-10-15 11:22:01.736 GMT [30] FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
2023-10-15 11:23:02.185 GMT [32] FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
2023-10-15 11:24:02.597 GMT [34] FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
2023-10-15 11:25:02.991 GMT [36] FATAL:  no pg_hba.conf entry for host "172.18.0.5", user "lemmy", database "lemmy", no encryption
```
My current **customPostgresql.conf** looks like this:
```
# DB Version: 15
# OS Type: linux
# DB Type: web
# Total Memory (RAM): 8 GB
# CPUs num: 4
# Data Storage: ssd

max_connections = 200
shared_buffers = 2GB
effective_cache_size = 6GB
maintenance_work_mem = 512MB
checkpoint_completion_target = 0.9
wal_buffers = 16MB
default_statistics_target = 100
random_page_cost = 1.1
effective_io_concurrency = 200
work_mem = 5242kB
huge_pages = off
min_wal_size = 1GB
max_wal_size = 4GB
max_worker_processes = 4
max_parallel_workers_per_gather = 2
max_parallel_workers = 4
max_parallel_maintenance_workers = 2
listen_addresses = '*'
```

**Comment by NachoPicchu** [comment]
### Solution:
The issue was that my customPostgresql.conf file did not contain the line 
`listen_addresses = '*'`
Everything else I've encountered is downstream from that. (I have resolved it by removing all docker containers and images **and removing the install directory with the volumes and config files**, running the uninstall.yml and installing again)

**Comment by NachoPicchu** [comment]
### Solution:

The issue was that my customPostgresql.conf file did not contain the line
`listen_addresses = '*'
`Everything else I've encountered is downstream from that. (I have resolved it by removing all docker containers and images **and removing the install directory with the volumes and config files**, running the uninstall.yml and installing again)

**Comment by codyro** [comment]
Looks like you got this figured out. This should be fixed in `1.2.1` (https://github.com/LemmyNet/lemmy-ansible/pull/183)
